homeassistant/components/buienradar/util.py

```diff
@@ -68,14 +68,10 @@ def __init__(self, hass, coordinates, timeframe, devices):
     async def update_devices(self):
         """Update all devices/sensors."""
         if self.devices:
-            tasks = []
             # Update all devices
             for dev in self.devices:
                 if dev.load_data(self.data):
-                    tasks.append(dev.async_update_ha_state())
-
-            if tasks:
-                await asyncio.wait(tasks)
+                    dev.async_write_ha_state()
```

Comment:
Let's keep it like this. Actually our service helper does the same: if entity.should_poll: request update.

Reply:
The service helper only uses already added entities, if I'm reading correctly. So that's fine. Normally the core is written so that it can safely use the entity api. It's the integrations where they access the entity from outside before the entity has been added, that is the problem.